content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,342 @@
+background are generated. In the second stage, a fit will be made for the
+signal and background. Here is how we can define the workflow steps:
+
+```python
+my_workflow = {
+    'steps': [
+        {
+            'name': 'gendata',
+            'environment': 'reanahub/reana-env-root6:6.18.04',
+            'commands': [
+                'mkdir -p results',
+                'root -b -q \'code/gendata.C(${events},"${data}")\' | tee gendata.log',
+            ],
+        },
+        {
+            'name': 'fitdata',
+            'environment': 'reanahub/reana-env-root6:6.18.04',
+            'commands': [
+                'root -b -q \'code/fitdata.C("${data}","${plot}")\' | tee fitdata.log'
+            ],
+        },
+    ]
+}
+```
+
+### Creating workflow
+
+Finally, once all the workflow steps are specified, we are ready to create it
+in REANA. To create the workflow using REANA client we could use the [create](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-create)
+command. In our case we can use `create_workflow_from_json` function
+where we need to define all the needed parameters:
+
+```python
+from reana_client.api.client import create_workflow_from_json
+
+create_workflow_from_json(
+    workflow_json=my_workflow,
+    name=my_workflow_name,
+    access_token=my_reana_token,
+    parameters=my_inputs,
+    workflow_engine=my_workflow_type)
+```
+
+Output:
+
+```python
+{
+  'message': 'Workflow workspace created',
+  'workflow_id': '6cd613eb-f2fb-411b-9601-c89599925759',
+  'workflow_name': 'root6-roofit'
+}
+```
+
+### Uploading files
+
+After creating our workflow we also need to upload all the input files to
+the workflow workspace. Same functionality can be achieved with REANA client
+[upload](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-upload)
+command. It can be done as follows:
+
+```python
+from reana_client.api.client import upload_to_server
+
+abs_path_to_input_files = [os.path.abspath(f) for f in my_inputs['files']]
+upload_to_server(my_workflow_name, abs_path_to_input_files, my_reana_token)
+```
+
+### Starting workflow
+
+At this point we have done all what is needed to start our workflow in REANA.
+To do it with REANA client we could call [start](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-start)
+command. But in this case it can be started by simply calling `start_workflow`
+function and defining workflow name and access token. Additionally, we
+can also override workflow parameters while starting the workflow, but let's
+keep the original ones for now:
+
+```python
+from reana_client.api.client import start_workflow
+
+start_workflow(my_workflow_name, my_reana_token, {})
+```
+
+Output:
+
+```python
+{
+  'message': 'Workflow submitted.',
+  'run_number': 1,
+  'status': 'queued',
+  'user': '1d27737b-5b80-4da2-87bb-816a20e110bd',
+  'workflow_id': '6cd613eb-f2fb-411b-9601-c89599925759',
+  'workflow_name': 'root6-roofit'
+}
+```
+
+### Checking workflow status
+
```

Comment:
We are now interested to check the workflow status. Is it waiting in the queue, it is running, has it failed ot finished? In command-line interface, we would use the `status` CLI command.. In Python, the corresponding function to use is `get_workflow_status()` function. Since our example workflow may take a few minutes to finish all the computations, we shall use a loop to print its status regularly after a certain wait time:

...

Example output:

...